Sources/Apodini/Semantic Model Builder/GraphQL/GraphQLSchemaBuilder.swift

```diff
@@ -0,0 +1,248 @@
+
+        if let typeVal = typeValTemp {
+            return typeVal as! GraphQLOutputType
+        }
+
+
+        // Array / Optional
+        if let newResponseHead = try! responseTypeHead.edited(handleArray)?.edited(handleOptional) {
+            if (newResponseHead.value.cardinality == .zeroToMany(.array)) { // Array
+                if let eNode = try? EnrichedInfo.node(newResponseHead.value.typeInfo.type) {
+                    return GraphQLList(responseTypeHandler(for: eNode))
+                }
+            } else if (newResponseHead.value.cardinality == .zeroToOne) { // Optional
+                if let eNode = try? EnrichedInfo.node(newResponseHead.value.typeInfo.type) {
+                    return responseTypeHandler(for: eNode)
+                }
+            }
+        }
+
+        var currentFields = [String: GraphQLField]()
+        for c in responseTypeHead.children {
+            if let propertyInfo = c.value.propertyInfo {
+                currentFields[propertyInfo.name] = GraphQLField(type: responseTypeHandler(for: c), resolve: { source, args, context, info in
+                    return try propertyInfo.get(from: source)
+                })
+            }
+        }
+
+        let typeName = responseTypeHead.value.typeInfo.name
+        return try! GraphQLObjectType(name: typeName, fields: currentFields)
+    }
+
+    private func graphQLFieldCreator(for responseTypeHead: Node<EnrichedInfo>, _ context: AnyConnectionContext<GraphQLInterfaceExporter>, _ args: [String: GraphQLArgument]) -> GraphQLField {
+        var mutableContext = context
+        let graphQLFieldType = self.responseTypeHandler(for: responseTypeHead)
+        return GraphQLField(type: graphQLFieldType, args: args, resolve: { gSource, gArgs, gContext, gEventLoop, gInfo in
+            let request = GraphQLRequest(source: gSource, args: gArgs, context: gContext, info: gInfo)
+            let vaporRequest = gContext as! Vapor.Request
+            let response: EventLoopFuture<Response<AnyEncodable>> = mutableContext.handle(request: request, eventLoop: vaporRequest.eventLoop.next())
+
+            return response.flatMapThrowing { encodableAction -> Any? in
+                switch encodableAction {
+                case let .send(element),
+                     let .final(element):
+                    return element.wrappedValue
+                case .nothing, .end:
+                    return ".nothing, .end"
+                }
+            }
+
+        })
+    }
+
+
+    // Generated adjacency list tree
+    func append<H: Handler>(for endpoint: Endpoint<H>, with context: AnyConnectionContext<GraphQLInterfaceExporter>) {
+        // Remove parameters from the path by using `":" filter`
+        var currentPath = endpoint.absolutePath.map {
+            $0.description.lowercased()
+        }.filter {
+            $0.first != ":"
+        }
+
+        // Remove `root`
+        currentPath.removeFirst()
+        // Remove `v1`
+        currentPath.removeFirst()
```

Comment:
I think that we should make this a bit more generic in the long term then just removing the first two path components as this might change in the future.